sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/SupportsOneDeploy.java

```diff
@@ -110,4 +110,20 @@ public interface SupportsOneDeploy {
      * if {@link DeployOptions#withTrackDeployment(Boolean)} set to True.
      */
     Mono<KuduDeploymentResult> pushDeployAsync(DeployType type, File file, DeployOptions deployOptions);
+
+    /**
+     * Gets the production deployment status for Azure Site.
+     *
+     * @param deploymentId GUID of the deployment operation
+     * @return the production deployment status for Azure Site
+     */
+    DeploymentBuildStatus getDeploymentStatus(String deploymentId);
```

Comment:
Why only a status `DeploymentBuildStatus`?

Why not the `CsmDeployment...`? Why remove the `Inner` but not use it?

Reply:
`CsmDeployment...` should've been the return type of another api `listDeploymentStatus`, but this api returns weird result(empty list) so currently I did not expose this api in convenience method.

Reply:
Ideally `getDeploymentStatus` should return same thing as `listDeploymentStatus`.

It is OK we do not expose `listDeploymentStatus` now. But I think other properties would also be useful (i.e. how many succeeded, how many failed, etc.).

Reply:
ok